PC Engines APU 2, 3, 4 & 6 v0.9.1 release branch

matmacieje opened the pull request:
*ref: sha-1492*

DonVasyl committed:
platform-configs/pcengines-apu*.robot: Configuration update for v0.9.1

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

DonVasyl committed:
USB & BPS test fixes for APU platforms

USB keyboard test in FW skipped
Windows BPS tests skipped

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

DonVasyl committed:
STB002.201 minimal-regression enabler

Not ideal, but works for now.

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

DonVasyl committed:
PC Engines APU: GNOME power control removed from Ansible Ubuntu playbook

DO NOT MERGE. We need either Ubuntu Server ENV_ID or some conditional
playbook control.

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

SergiiDmytruk commented on dasharo-compatibility/usb-hid-and-msc-support.robot:
```diff
@@ -42,6 +42,7 @@ USB002.001 USB keyboard detected in FW
     [Tags]    automated    minimal-regression
     Depends On    ${TESTS_IN_FIRMWARE_SUPPORT}
     Depends On    ${HAS_KEYBOARD}
+    Skip If    '${MANUFACTURER}' == 'PC Engines'
```
This test depends on `${HAS_KEYBOARD}` being true.  It's set to true in `Prepare USB HID Test Suite` keyword if `"${DEVICE_USB_KEYBOARD}" != "${EMPTY}" or "${DUT_CONNECTION_METHOD}" == "pikvm"`.  I think both conditions are false, so how come this test isn't already skipped?

matmacieje replied:
Removed in https://github.com/Dasharo/open-source-firmware-validation/pull/1158/commits/c6466471e60dc66a6b22bdaa922e9bb7bb08b8f4

SergiiDmytruk commented on util/basic-platform-setup.robot:
```diff
@@ -111,7 +111,13 @@ BPS006.001 Ensure test dependencies
 BPS007.001 External flashing
     [Documentation]    This test verifies if the flash die can be detected.
     Skip If    '${FLASHING_METHOD}' != 'external'
+    IF    '${MANUFACTURER}' == 'PC Engines' and '${APU_FLASH_WP_GPIO}' != '${TBD}'
```
I think conditions can be simplified down to:
```robot
    IF    '${APU_FLASH_WP_GPIO}' != '${TBD}'
```

matmacieje replied:
Changed in https://github.com/Dasharo/open-source-firmware-validation/pull/1158/commits/c6466471e60dc66a6b22bdaa922e9bb7bb08b8f4

DonVasyl committed:
platform-configs/include/pcengines.robot: keyboard config updated

PC Engines-specific conditions removed from:

USB keybard test skip:
dasharo-compatibility/usb-hid-misc-msc-support.robot

Flash WP GPIO manipulation:
util/basic-platform-setup.robot
lib/flash.robot

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

DonVasyl committed:
os-config/ansible/linux-packages-playbook.yaml: changes reverted.

Ubuntu Server-enabling chnages reverted, issue created:
https://github.com/Dasharo/open-source-firmware-validation/issues/1159

Changes & log in attachment.

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

SergiiDmytruk commented:
@matmacieje, the third commit says:
```
STB002.201 minimal-regression enabler

Not ideal, but works for now.
```
What's problematic about it?  No way to specify `minimal-regression` tag and an extra test at the same time?

matmacieje commented:
@SergiiDmytruk this is quick fix proposed by @philipanda 

With introduction of concurrent test execution, minimal-regression tag has been removed but also become incompatible with new architecture. Adding it like this is possible only for shortest test cases and some redesign is required in libs and/or concurrent suites to cover everything.

philipanda commented:
> @SergiiDmytruk this is quick fix proposed by @philipanda
> 
> With introduction of concurrent test execution, minimal-regression tag has been removed but also become incompatible with new architecture. Adding it like this is possible only for shortest test cases and some redesign is required in libs and/or concurrent suites to cover everything.

Made an issue for that https://github.com/Dasharo/open-source-firmware-validation/issues/1160
